src/components/Layout/TableOfContents.jsx

```diff
@@ -1,165 +1,90 @@
-import React from "react"
-import Link from 'gatsby-link'
-import styled from 'styled-components'
-
-
-class TableOfContents extends React.Component {
```

Comment:
Only use classes when you need state. You do not need state here or in Gatsby in general. So use stateless (aka functional) components. It makes everything so much easier.